week1/homework/src/server.js

```diff
@@ -1,20 +1,33 @@
 'use strict';
 
 const http = require('http');
+const calculation = require('./calculations');
 
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
 function createServer(port) {
-  let state = 10;
+  // let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    if (request.url === '/') {
```

Comment:
You are going over the `path` twice now: once here and once in `calculations.js`. 
You should improve your code to go only over the `path` of the url once. Otherwise, if you ever want to add another route, you would have to do this in two places. That would be fine if you would have to do this in - say - a week, but if someone else would have to do this, or you have to do this in a year, or if this were part of a big application, it would be difficult, and one of the instances could be forgotten. 
This is part of the DRY (Don't Repeat Yourself) principle: you want to keep repeating code to a minimum to reduce the chance someone makes a mistake in one of the copies of the code. 
You can pass the `request.url` directly to `calculation.calculator()`.

Reply:
Thanks Josja. I will go through my code again. Your comments are really useful.